### DISCUSSION TITLE
leptos is great, but how should it be deployed？

### DISCUSSION THREAD

**Discussion by amar-yaozhian** [leptos-rs/leptos#1149] [Q&A]
leptos is the best web framework I've seen: one-stop, high-performance...
But how to deploy is the most beautiful？

**Comment by gbj** [comment]
Leptos with SSR/hydration is designed to plug into common Rust server frameworks so the answer is basically "however you deploy your Actix/Axum" apps. Here's [the Dockerfile for the Leptos website](https://github.com/leptos-rs/leptos-website/blob/main/Dockerfile) for example. (It's deployed by GitHub Action onto Fly.io) It's mostly a matter of just building the server binary and making sure that the necessary static files can be served.

For a CSR app you can just deploy the files Trunk puts into `dist`.
